lib/inputstreamhelper/__init__.py

```diff
@@ -133,7 +133,11 @@ def _diskspace(cls):
     @classmethod
     def _temp_path(cls):
         ''' Return temporary path, usually ~/.kodi/userdata/addon_data/tmp '''
-        temp_path = os.path.join(ADDON_PROFILE, 'tmp')
+        cust_temp_dir = ADDON.getSetting('custom_temp_dir')
+        if xbmcvfs.exists(cust_temp_dir):
+            temp_path = os.path.join(cust_temp_dir, 'tmp')
+        else:
+            temp_path = os.path.join(ADDON_PROFILE, 'tmp')
```

Comment:
Should we not just create the directories, rather than silently not using it.

Reply:
this is meant for if a removable device is set, but no longer there.

Reply:
@dagwieers The user selected the custom temporary directory from the GUI, it definitely exists. The standard temporary directory is created if it doesn't exist. I don't see the problem here.